app/main.py

```diff
@@ -1,2 +1,82 @@
+from typing import Hashable, Any
+
+
 class Dictionary:
-    pass
+    LOAD_FACTOR = 2 / 3
+
+    def __init__(self) -> None:
+        self.size = 0
+        self.capacity = 8
+        self.hash_table = [None] * self.capacity
+
+    def _resize(self) -> None:
+        self.capacity *= 2
+        self.size = 0
+        old_table = self.hash_table
+        self.hash_table = [None] * self.capacity
+        for key, value in filter(None, old_table):
+            self.__setitem__(key, value)
+
+    def _get_index(self, key: Hashable) -> Any:
+        index = hash(key) % self.capacity
+        while True:
+            if (self.hash_table[index] is None
+                    or self.hash_table[index][0] == key):
+                return index
+            index = (index + 1) % self.capacity
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index = self._get_index(key)
```

Comment:
It already returns the index which we need, as for me. Maybe inside we do not need while loop?